[ADD] mrp_subcontract_location

florian-dacosta opened the pull request:
Still need to add tests and why not some demo data

@chafique-delli @hparfr

florian-dacosta committed:
[ADD] mrp_subcontract_location

hparfr committed:
update readme

hparfr commented on mrp_subcontract_location/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+    'author': 'Akretion',
+    'website': 'www.akretion.com',
+    'license': 'AGPL-3',
+    'category': 'Generic Modules',
```
Catégorie Manufacturing

florian-dacosta replied:
Avant de faire des review aux autres, je devrais déjà me faire la même à moi même!

hparfr commented on mrp_subcontract_location/__manifest__.py:
```diff
@@ -0,0 +1,22 @@
+    'license': 'AGPL-3',
+    'category': 'Generic Modules',
+    'description': """
+        Manage stock by location for subcontract manufacture order
```
subcontracted

hparfr commented on mrp_subcontract_location/models/mrp_production.py:
```diff
@@ -0,0 +1,49 @@
+#
+#  licence AGPL version 3 or later
+#  see licence in __openerp__.py or http://www.gnu.org/licenses/agpl-3.0.txt
+#  Copyright (C) 2015 Akretion (http://www.akretion.com).
```
2015?

hparfr commented on mrp_subcontract_location/models/mrp_production.py:
```diff
@@ -0,0 +1,49 @@
+        subcontract_main_loc = subcontract_wh.lot_stock_id
+        if self.location_src_id.id != subcontract_main_loc.id or \
+                self.location_dest_id.id != subcontract_main_loc.id:
+            return
```
j'ai cru à un retour a la ligne malencontreux

florian-dacosta replied:
C'est à dire?

hparfr replied:
j'ai cru que tu voulais écrire `return location_obj = self.env...`. 
Pas spécialement besoin de corriger.

hparfr commented on mrp_subcontract_location/models/mrp_production.py:
```diff
@@ -0,0 +1,49 @@
+        if self.location_src_id.id != subcontract_main_loc.id or \
+                self.location_dest_id.id != subcontract_main_loc.id:
+            return
+        location_obj = self.env['stock.location']
```
code mort ?

hparfr commented on mrp_subcontract_location/models/purchase_order.py:
```diff
@@ -0,0 +1,40 @@
+        for purchase in self:
+            for line in purchase.order_line:
+                if line._is_service_procurement():
+                    mo = line.procurement_ids and \
```
no need to test on line.procurement_ids if it's a service_procurement()

florian-dacosta replied:
le _is_service_procurement vérifie juste que le produit est de type service et avec la case subcontracted service cochée, mais si on ajoute un produit de ce genre manuellement dans une PO, elle aura ni procurement, ni MO de liée ?

hparfr replied:
is_service_**procurement**__  le nom implique un procurement donc faudrait faire ce check dans cette méthode.

(d'ailleurs on devrait peut être renommer en is_procurement_service),

hparfr replied:
https://github.com/akretion/ak-odoo-incubator/pull/87/commits/0fa95efb7556793850a96cf171af4e137f347d4d

florian-dacosta replied:
J'ai corrigé en anticipant ton fix sur "_is_service_procurement"

florian-dacosta committed:
Fix some details

florian-dacosta committed:
Fix location on picking where there is a supplier destination filled

hparfr committed:
Change location and makes new pickings

hparfr committed:
Manage multiple outgoing products

hparfr committed:
Move dates

hparfr committed:
One picking for one PO

hparfr committed:
Add default values

hparfr committed:
Refactorings

Pickings are now grouped

hparfr committed:
split purchase order line

florian-dacosta committed:
If warehouse is default stock one assign default supplier to MO

florian-dacosta committed:
Allow make to stock + handle customer delivery case (last step)

florian-dacosta committed:
Big refactore to create all reception/delivery moves between MOs in advance, using rules

florian-dacosta committed:
Fix picking group behavior

florian-dacosta committed:
Refactore

hparfr committed:
Assign partners in order to have them directly in the pickings

hparfr committed:
Add cut

hparfr committed:
Add cut

florian-dacosta committed:
Fix qty received qty for subcontracted mo purchase order

hparfr committed:
Remove unneeded assign

 moves and not update at every approve

hparfr committed:
Fix scrap : location don't work by default in v10
+ unreserve next moves after scrap (in order to update transfert quantities)

hparfr committed:
Revert "Add cut"

This reverts commit fd551aff062bd4911aecd4577bbb88fc6ec1a0ea.

hparfr committed:
Merge pull request #2 from florian-dacosta/10-mrp_subcontract_location_wip_assign_partners_on_rules

Assign partners in order to have them directly in the pickings

florian-dacosta committed:
Merge pull request #3 from florian-dacosta/10-mrp_subcontract_location_wip_scrap

Fix scrap : location don't work by default in v10

florian-dacosta committed:
Add comment about make to order case (not managed)

florian-dacosta committed:
Merge pull request #4 from florian-dacosta/10-mrp_subcontract_location_wip_uncut

10 mrp subcontract location wip uncut

florian-dacosta committed:
Refactore

florian-dacosta committed:
Compute product routes from boms + get rid of location_id field on supplier (#118)

hparfr committed:
Refactoring

hparfr committed:
Fix small edge cases

hparfr committed:
Manage warehouse partner as child of seller
Change info log message to warning

hparfr committed:
Manage child location

hparfr committed:
Fix bug with domain

hparfr committed:
Split picking per partner

hparfr committed:
Add fix search_suitable_rule

Add receiver address in the search

Fall back if it returns no result

hparfr committed:
Fix typo

hparfr committed:
Fix @api.multi

hparfr committed:
Remove uneeded code, we now use picking_type_id as default location of a bom

hparfr committed:
Pylint

hparfr committed:
Run cron differentrly

hparfr committed:
Don't move date anymore

hparfr committed:
Write po.picking_type on confirm of po
Remove dead code
Imp readme

hparfr committed:
Use pol.mo_id instead of pol.procurement_id.production_id

hparfr committed:
externalize route creation

hparfr committed:
Major change: raw move ids are no longer moved between suppliers

Raw materials moves are cancelled
New raw materials moves are generated at the new location

hparfr committed:
Add tests

hparfr committed:
Group pickings per SO

hparfr committed:
Fix buy products

hparfr committed:
Fix cancelation

hparfr committed:
Always try to choose a route with a partner_address_id
and fix tests import

hparfr committed:
Fix bug when two mo at the same place + test

chafique-delli committed:
[FIX] assign_picking in stock.move

chafique-delli committed:
Merge pull request #214 from akretion/10-fix-mrp_subcontract_location

[FIX] assign_picking in stock.move

chafique-delli committed:
[FIX] assign_picking method when stock move is done

chafique-delli committed:
Merge pull request #238 from akretion/10-fix-mrp_subcontract_location

[FIX] assign_picking method when stock move is done

chafique-delli committed:
[FIX] update_moves_after_production method when destination move is done

chafique-delli committed:
Merge pull request #241 from akretion/10-fix-mrp_subcontract_location

[FIX] update_moves_after_production method when destination move is done

chafique-delli committed:
[FIX] assign_picking method when empty_picking is Available

chafique-delli committed:
create a method _check_same_wh() in mrp.bom for check if the Product and raw material are manufactured in the same wh in order to overload it

chafique-delli committed:
also set routes on products whose BOM only has purchased components

chafique-delli committed:
Merge pull request #286 from akretion/10-fix-mrp_subcontract_location_wip

also set routes on products whose BOM only has purchased components
